tt_metal/Glossary.md

```diff
@@ -0,0 +1,110 @@
+- **LUT -** Look Up Table - Mapping table of input and output values used for addressing.
+- **MLP -** Multilayer Perceptron - A modern feedforward artificial neural network.
+- **MLIR -** Multi-Level Intermediate Representation
+- **MM - MatMul - Matrix Multiplication -** Multiplication of two square or rectangular matrices.
+- **MIMD -** Multiple Instruction Multiple Data - Allows multiple processors to function simultaneously and asynchronously.
+- **NM -** Nanometers
+- **NMC -** Near Memory Computing - Computing paradigm that moves data processing closer to where the data is stored.
+- **NLP -** Natural Language Processing
+- **NoC -** Network on Chip
+- **NPU -** Neural Processing Unit
+- **NUMA -** Non-uniform Memory Access
+- **PCC -** Pearson Correlation Coefficient - A method of measuring a linear correlation between two variables.
+- **Q, K, V values -** Q, K, and V stand for different values. Q = query, K = key, V = value. In transformer architecture these values are used to help the model focus on relevant parts of the input sequence.
+- **SFPU -** Vetor unit.
+- **SIMD -** Single Instruction Multiple Data - Allows processors to execute the same instruction on multiple data points simultaneously
+- **SoC -** System on Chip
+- **SRAM -** Static Random Access Memory. On-chip Memory. 1 MB SRAM memory (L1), a scratch pad accessible by all RISC-V processors and engines within the core.
+- **TM -** Tensor Manipulation
+- **TP -** Tensor Parallel
+- **ViT -** Vision Transformer - A transformer model used for vision processing tasks.
+## **Terminology** ##
+- **Convolution -** Multiplication of two functions’ values to create a new function.
+- **DPRINT -** Configurable code formatting platform.
+- **dtype -** Data Type.
+- **Huge Page -** Memory management technique used to request larger memory blocks.
+- **Kernel Types -** Kernels send and receive data from tensors.
+  - **Bare Metal C/C++ Kernels -** Kernels written in C or C++ for execution. Generally used to accelerate computation and processing performance.
+  - **Reader Kernel -** Kernel receiving data from a DRAM or SRAM buffer.
+  - **Writer Kernel -** Kernel sending data to a DRAM or SRAM buffer.
+  - **User Kernel Types -** Kernels are in user mode when operating in applications.
+    - **Compute Kernels -** Kernels used for processing tasks or operations. Compute kernels will automatically generate the following types of kernels:
+      - **Unpack Kernels -** Unpack kernels prepare data for operations to be performed by the math kernel.
+      - **Math Kernels -** Kernels used for matrix multiplication and other mathematical tasks or operations.
+      - **Pack Kernels -** Pack kernels wait for the end of the math kernel and prepare data to be moved to the next part of the system.
+    - **Data Movement Kernels _** The first and fifth RISC-Vs on a Tensix Core responsible for moving data between NoCs, memory buffers, and the compute kernel.
```

Comment:
Please make better hierarchy. Read/write kernels are data movement. All of them (including the 3 compute) are user kernels.